addons/mail/static/src/models/thread/thread.js

```diff
@@ -633,23 +666,14 @@ function factory(dependencies) {
          *
          * Only makes sense if isPendingPinned is set to the desired value.
          */
-        notifyPinStateToServer() {
-            // method is called from _updateAfter so it cannot be async
+        async notifyPinStateToServer() {
             if (this.isPendingPinned) {
-                this.env.services.rpc({
-                    model: 'mail.channel',
-                    method: 'channel_pin',
-                    kwargs: {
-                        uuid: this.uuid,
-                        pinned: true,
-                    },
-                }, { shadow: true });
+                await this.env.models['mail.thread'].performRpcChannelPin(this.uuid, true);
```

Comment:
I guess object notation (instead of position) for params would make this more clear. Hard to guess what `true` is about here.